### PR TITLE
Storage test

### PR DESCRIPTION
**Pull request by alelb**
This PR introduces storage tests with the new document structure.

### CODE REVIEW

File: storage/mongo/adapter_test.go [hunk 1]
```diff
@@ -165,17 +169,17 @@ func TestConvertToMongoQuery(t *testing.T) {
...
-			require.NoError(t, err, herror(ii, tt))
+	// 		m, err := convertToMongoQuery(&res, ft)
+	// 		require.NoError(t, err, herror(ii, tt))
 
-			require.Equal(t, tt.query1, m, herror(ii, tt))
-		})
-	}
+	// 		require.Equal(t, tt.query1, m, herror(ii, tt))
+	// 	})
+	// }
```
Comment (leogr) [inline storage/mongo/adapter_test.go]: WIP?

File: storage/mongo/adapter_test.go [hunk 2]
```diff
@@ -14,139 +14,143 @@ import (
...
-				``,
-			},
-			{
-				`not (userName eq "strings")`,
-				``,
-				``,
-			},
-	*/
+		//bson.M{"$or": []interface{}{bson.M{"$and": []interface{}{bson.M{"_": bson.M{"$eq": "work"}}, bson.M{"_": bson.M{"": "@example.com"}}}}, bson.M{"$and": []interface{}{bson.M{"_": bson.M{"$eq": "xmpp"}}, bson.M{"_": bson.M{"": "@foo.com"}}}}}, "meta.resourceType": "User"},
+		bson.M{"$or": []interface{}{bson.M{"$and": []interface{}{bson.M{"urn:ietf:params:scim:schemas:core:2°0:User.emails.type": bson.M{"$eq": "work"}}, bson.M{"urn:ietf:params:scim:schemas:core:2°0:User.emails.value": bson.M{"$regex": bson.RegEx{Pattern: "@example\\.com", Options: "i"}}}}}, bson.M{"$and": []interface{}{bson.M{"urn:ietf:params:scim:schemas:core:2°0:User.ims.type": bson.M{"$eq": "xmpp"}}, bson.M{"urn:ietf:params:scim:schemas:core:2°0:User.ims.value": bson.M{"$regex": bson.RegEx{Pattern: "@foo\\.com", Options: "i"}}}}}},
```
Comment (leogr) [inline storage/mongo/adapter_test.go]: @leodido May be convenient to escape keys by using `keyEscape()` inline, instead of using `°` directly?
cc @alelb

**Comment by alelb** [comment]
cc @leogr @leodido please review PR

**Review by leodido — APPROVED** [review]
Excellent.

**Pull request merged**